bin/stress-test/src/main.rs

```diff
@@ -0,0 +1,287 @@
+        .with_component(
+            BasicFungibleFaucet::new(TokenSymbol::new("TEST").unwrap(), 2, Felt::new(100_000))
+                .unwrap(),
+        )
+        .build()
+        .unwrap();
+    new_faucet.id()
+}
+
+/// Create a new note containing 10 tokens of the fungible asset associated with the specified
+/// `faucet_id`.
+fn create_note(faucet_id: AccountId) -> Note {
+    let asset = Asset::Fungible(FungibleAsset::new(faucet_id, 10).unwrap());
+    let coin_seed: [u64; 4] = rand::thread_rng().gen();
+    let rng = RpoRandomCoin::new(coin_seed.map(Felt::new));
+    NoteBuilder::new(faucet_id, rng)
+        .add_assets(vec![asset])
+        .build(&TransactionKernel::assembler())
+        .unwrap()
+}
+
+/// Create a new account with a given public key and anchor block. Generates the seed from the given
+/// index.
+fn create_account(anchor_block: &BlockHeader, public_key: PublicKey, index: u64) -> AccountId {
+    let init_seed: Vec<_> = index.to_be_bytes().into_iter().chain([0u8; 24]).collect();
+    let (new_account, _) = AccountBuilder::new(init_seed.try_into().unwrap())
+        .anchor(anchor_block.try_into().unwrap())
+        .account_type(AccountType::RegularAccountImmutableCode)
+        .storage_mode(AccountStorageMode::Private)
+        .with_component(RpoFalcon512::new(public_key))
+        .with_component(BasicWallet)
+        .build()
+        .unwrap();
+    new_account.id()
+}
+
+/// Build blocks from transaction batches. Each new block contains [`BATCHES_PER_BLOCK`] batches.
+/// Returns the total time spent on inserting blocks to the store and the number of inserted blocks.
+async fn build_blocks(
+    mut batch_receiver: UnboundedReceiver<TransactionBatch>,
+    store_client: StoreClient,
+) -> (Duration, u32) {
+    let block_builder = BlockBuilder::new(store_client);
+
+    let mut current_block: Vec<TransactionBatch> = Vec::with_capacity(BATCHES_PER_BLOCK);
+    let mut insertion_times = Vec::new();
+    while let Some(batch) = batch_receiver.recv().await {
+        current_block.push(batch);
+
+        if current_block.len() == BATCHES_PER_BLOCK {
+            let start = Instant::now();
+            block_builder.build_block(&current_block).await.unwrap();
+            insertion_times.push(start.elapsed());
+            current_block.clear();
+        }
+    }
+
+    if !current_block.is_empty() {
+        let start = Instant::now();
+        block_builder.build_block(&current_block).await.unwrap();
+        insertion_times.push(start.elapsed());
+    }
+
+    let num_insertions = insertion_times.len() as u32;
+    let insertion_time: Duration = insertion_times.iter().sum();
+    (insertion_time, num_insertions)
+}
+
+/// Create a given number of notes and group them into transactions and batches.
+/// The batches are sent to the block builder.
+fn generate_note_batches(
+    num_notes: usize,
+    faucet_id: AccountId,
+    batch_sender: UnboundedSender<TransactionBatch>,
+) -> Vec<Note> {
```

Comment:
I'm not sure I fully understood how this works. Wouldn't we need to create just 1 transaction per block to create notes? That is, this transaction would be executed against the faucet account and would output 256 notes directed to 256 accounts that were created in the previous block. Or does this cause issues somehow?